src/NLog/LayoutRenderers/ExceptionLayoutRenderer.cs

```diff
@@ -229,8 +230,12 @@ protected override void Append(StringBuilder builder, LogEventInfo logEvent)
 #if !NET3_5 && !SILVERLIGHT4
                 if (logEvent.Exception is AggregateException aggregateException)
                 {
+                    var aggregateExceptionData = aggregateException.Data;
                     aggregateException = aggregateException.Flatten();
                     primaryException = GetPrimaryException(aggregateException);
+                    foreach (var key in aggregateExceptionData.Keys)
+                        primaryException.Data.Add(key, aggregateExceptionData[key]);
```

Comment:
Prefer that the logging-library only probes the Exception, and doesn't modify the Exceptions.